File: src/main/resources/static/js/class/Cliente.js



export class Cliente {
    constructor({
        idCliente = null,
        idPropietario = null,
        telefono = null,
        nombre = null,
        apellidoPaterno = null,
        apellidoMaterno = null,
        idColonia = null,
        numeroExterior = null,
        calle = null,
        fechaAlta = null,
    } = {})
    {
        this.idCliente = idCliente;
        this.idPropietario = idPropietario;
        this.telefono = telefono;
        this.nombre = nombre;
        this.apellidoPaterno = apellidoMaterno;
        this.apellidoMaterno = apellidoMaterno;
        this.idColonia = idColonia;
        this.numeroExterior = numeroExterior;
        this.calle = calle;
        this.fechaAlta = fechaAlta;
    }
    
    get getId() {
        return this.id;
    }
    set setId(id) {
        this.id = id;
    }
    get getNombre() {
        return this.nombre;
    }
    set setNombre(nombre) {
        this.nombre = nombre;
    }
    get getApellidoPaterno() {
        return this.apellidoPaterno;
    }
    set setApellidoPaterno(apellidoPaterno) {
        this.apellidoPaterno = apellidoPaterno;
    }
    get getApellidoMaterno() {
        return this.apellidoMaterno;
    }
    set setApellidoMaterno(apellidoMaterno) {
        this.apellidoMaterno = apellidoMaterno;
    }
    get getCorreo() {
        return this.correo;
    }
    set setCorreo(correo) {
        this.correo = correo;
    }
    get getTelefono() {
        return this.telefono;
    }
    set setTelefono(telefono) {
        this.telefono = telefono;
    }
    get getDireccion() {
        return this.direccion;
    }
    set setDireccion(direccion) {
        this.direccion = direccion;
    }
    get getColonia() {
        return this.colonia;
    }
    set setColonia(colonia) {
        this.colonia = colonia;
    }
    get getNumeroExterior() {
        return this.numeroExterior;
    }
    set setNumeroExterior(numeroExterior) {
        this.numeroExterior = numeroExterior;
    }
}

File: src/main/java/com/vs/alafe/model/entities/Cliente.java
package com.vs.alafe.model.entities;

import com.vs.alafe.model.interfaces.ALAFEEntity;
import jakarta.persistence.*;
import org.antlr.v4.runtime.misc.NotNull;
import java.io.Serializable;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;
import java.util.stream.Collectors;
import java.util.stream.Stream;

@Entity
@Table(name = "cliente")
public class Cliente implements Serializable{

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_cliente",nullable = false)
    private Integer idCliente;
    @JoinColumn(name = "id_propietario",nullable = false)
    @ManyToOne(fetch = FetchType.LAZY)
    private Propietario propietario;
    @Column(name = "telefono")
    private String telefono;
    @Column(name = "nombre")
    private String nombre;
    @Column(name = "apellido_paterno")
    private String apellidoPaterno;
    @Column(name = "apellido_materno")
    private String apellidoMaterno;
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "id_colonia", nullable = true)
    private Colonia colonia;
    @Column(name = "numero_exterior")
    private String numeroExterior;
    @Column(name = "calle")
    private String calle;
    @Column(name = "fecha_alta")
    private LocalDateTime fechaAlta;
    @Column(name = "correo")
    private String correo;

    @OneToMany(mappedBy = "cliente", fetch = FetchType.LAZY)
    private List<Evento> eventos = new ArrayList<>();


    public Cliente() {
    }

    public Integer getIdCliente() {
        return idCliente;
    }

    public void setIdCliente(Integer idCliente) {
        this.idCliente = idCliente;
    }

    public Propietario getPropietario() {
        return propietario;
    }

    public List<Evento> getEventos() {
        return eventos;
    }

    public void setEventos(List<Evento> eventos) {
        this.eventos = eventos;
    }

    public void setPropietario(Propietario propietario) {
        this.propietario = propietario;
    }

    public String getTelefono() {
        return telefono;
    }

    public void setTelefono(String telefono) {
        this.telefono = telefono;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getApellidoPaterno() {
        return apellidoPaterno;
    }

    public void setApellidoPaterno(String apellidoPaterno) {
        this.apellidoPaterno = apellidoPaterno;
    }

    public String getApellidoMaterno() {return apellidoMaterno;}

    public void setApellidoMaterno(String apellidoMaterno) {this.apellidoMaterno = apellidoMaterno;}
    public Colonia getColonia() {
        return colonia;
    }

    public void setColonia(Colonia idColonia) {
        this.colonia = idColonia;
    }


    public String getNumeroExterior() {
        return numeroExterior;
    }

    public void setNumeroExterior(String numeroExterior) {
        this.numeroExterior = numeroExterior;
    }

    public String getCalle() {
        return calle;
    }

    public void setCalle(String calle) {
        this.calle = calle;
    }

    public LocalDateTime getFechaAlta() {
        return fechaAlta;
    }

    public void setFechaAlta(LocalDateTime fechaAlta) {
        this.fechaAlta = fechaAlta;
    }

    public String getCorreo() {
        return correo;
    }

    public void setCorreo(String correo) {
        this.correo = correo;
    }

    public String nombreCompleto() {
        return Stream.of(nombre, apellidoPaterno, apellidoMaterno)
                .filter(s -> s != null && !s.isEmpty())
                .collect(Collectors.joining(" "));
    }

    @Override
    public String toString() {
        return "Cliente{" +
                "idCliente=" + idCliente +
                ", telefono='" + telefono + '\'' +
                ", nombre='" + nombre + '\'' +
                ", apellidoPaterno='" + apellidoPaterno + '\'' +
                ", apellidoMaterno='" + apellidoMaterno + '\'' +
                ", colonia=" + colonia +
                ", numeroExterior='" + numeroExterior + '\'' +
                ", calle='" + calle + '\'' +
                ", fechaAlta=" + fechaAlta +
                ", correo=" + correo +
                '}';
    }
}
